### PR TITLE
Hooks should run with their pod config as well

### PR DESCRIPTION
**Pull request by anthonybishopric**

### CODE REVIEW

File: pkg/runit/runit_service.go [hunk 1]
```diff
@@ -67,3 +68,7 @@ func convertToErr(msg string, original error) (string, error) {
 		return msg, original
 	}
 }
+
+func (service *Service) RunScript() string {
```
Comment (anthonybishopric) [inline pkg/runit/runit_service.go]: This is unused - will remove

**Comment by tummychow** [comment]
I'm going to roll this branch out in a universe first, but code lgtm.

**Comment by tummychow** [comment]
Oh, one thing I forgot to mention but just remembered: hook_env.go should be changed to take advantage of the new environment variables.

**Comment by tummychow** [comment]
okay, did another pass. Code lgtm except for the three tasks as I mentioned, but to summarize:

- hook environment appending to existing environment vs creating a new one (should probably be the latter so that the preparer's environment is not inherited)
- hook_env.go should have functions for the new variables, and existing functions should be ported onto the new variables if they haven't been already
- we have `pkg/config.LoadFromEnvironment` for `CONFIG_PATH`, we should have something similar for `HOOKED_CONFIG_PATH` (could be in pkg/hooks or pkg/config, whichever one makes more sense to you)

Functionality also lgtm. Once the comments are addressed I will take a third pass and probably a fourth to be sure I didn't miss anything.

**Comment by tummychow** [comment]
code lgtm, gonna take it for another spin to confirm functionality again

**Comment by anthonybishopric** [comment]
The travis failure is real this time. Looking into it.

File: pkg/pods/hoist_executable_test.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	"strings"
+	"testing"
+
+	. "github.com/anthonybishopric/gotcha"
+)
+
+func TestExecutableWritesValidScript(t *testing.T) {
+	user, err := user.Current()
+	Assert(t).IsNil(err, "test setup failure - should not have failed to get current user")
+	envdir, err := ioutil.TempDir("", "envdir")
```
Comment (petertseng) [inline pkg/pods/hoist_executable_test.go]: the error from this line is never checked because the following line immediately overwrites it

Comment (petertseng) [inline pkg/pods/hoist_executable_test.go]: eh I guess it'll be fine because the WriteFile into the directory will fail

Comment (anthonybishopric) [inline pkg/pods/hoist_executable_test.go]: I'm also making a (maybe dumb) assumption that the next line, which is also a temp directory creation, would almost definitely have an error if the previous line had an error.

**Pull request merged**